django/apps/existing_database/query.py

```diff
@@ -0,0 +1,259 @@
+    """
+    with connections[settings.MAPSWIPE_EXISTING_DB].cursor() as cursor:
+        cursor.execute(COMMUNITY_DAHBOARD_STATS_QUERY)
+        aggregate_results = cursor.fetchall()
+    for data in aggregate_results:
+        return CommunityStatsType(
+            total_contributors=data[1] or 0,
+            total_groups=data[0] or 0,
+            total_swipes=data[2] or 0,
+        )
+
+
+def get_community_stats_latest() -> CommunityStatsLatestType:
+    COMMUNITY_DAHBOARD_STATS_QUERY = f"""
+        WITH
+            -- Group by project_id, group_id, user_id and user_group_id to get
+            -- Max start_time and end_time as they are repeated for each task,
+            -- but the value is at project_id, group_id, user_id level
+            dashboard_data AS (
+                SELECT
+                    UGR.user_group_id as user_group_id,
+                    R.user_id as user_id,
+                    R.timestamp as timestamp,
+                    COUNT(*) swipe_count
+                From {Result._meta.db_table} R
+                    LEFT JOIN {UserGroupResult._meta.db_table} UGR USING (user_id)
+                WHERE timestamp::date >= (CURRENT_DATE- INTERVAL '30 days')
+                GROUP BY user_group_id, user_id, timestamp
+            )
+        SELECT
+            COUNT( DISTINCT user_group_id) as total_groups,
+            COUNT(DISTINCT user_id) as total_users,
+            SUM(swipe_count) as total_swipes
+        From dashboard_data
+    """
+    with connections[settings.MAPSWIPE_EXISTING_DB].cursor() as cursor:
+        cursor.execute(COMMUNITY_DAHBOARD_STATS_QUERY)
+        aggregate_results = cursor.fetchall()
+    for data in aggregate_results:
+        return CommunityStatsLatestType(
+            total_contributors_last_month=data[1] or 0,
+            total_groups_last_month=data[0] or 0,
+            total_swipes_last_month=data[2] or 0,
+        )
+
+
+def get_contributor_time_stats() -> List[ContributorTimeType]:
+    COMMUNITY_DASHBOARD_STATS_QUERY = f"""
+        WITH
+        user_data AS (
+            SELECT
+                MAX(R.start_time) as start_time,
+                MAX(R.end_time) as end_time,
+                R.timestamp as timestamp
+            From {Result._meta.db_table} R
+            GROUP BY timestamp
+        )
+    SELECT
+        timestamp::date as task_date,
+        SUM(
+            EXTRACT(
+                EPOCH FROM (end_time - start_time)
+            )
+        ) as total_time
+    From user_data
+    GROUP BY timestamp
+    """
+    with connections[settings.MAPSWIPE_EXISTING_DB].cursor() as cursor:
+        cursor.execute(COMMUNITY_DASHBOARD_STATS_QUERY)
+        aggregate_results = cursor.fetchall()
+    list_data = []
+    for data in aggregate_results:
+        list_data.append(
+            ContributorTimeType(total_time=round(data[1] / 60) or 0, task_date=data[0])
+        )
+    return list_data
+
+
+def get_project_type_query() -> List[ProjectTypeStats]:
```

Comment:
I'm not sure if this function name is clear. As I understand it you are running a query for the build_area project type here not for all project types. I would suggest to change the name.

Reply:
There was a lot of inconsistency in the naming. We have changed a lot. Can you re-check? Sorry about this.